### PR TITLE
Fix: env0_agent_values data source doesn't fetch values

### LINKED ISSUE #690: `env0_agent_values` data source doesn't fetch values
Motivation / Problem / The Why❔:

Getting agent details using the [env0_agent_values](https://registry.terraform.io/providers/env0/env0/latest/docs/data-sources/agent_values) doesn't work

How to reproduce:

Run the following TF file, provide `env0_api_key_id`, `env0_api_key_value` and `http_auth` as TF_VAR. 
`http_auth` is `Basic <the api key and value encoded in base64>`, use the widget in our API reference to calculate it:
image

```

locals {
  api_endpoint    = <fill>
  kushield_env_org_id = <fill>
  agent_key = "my-agent-key"
}

variable "env0_api_key_id" {}
variable "env0_api_key_value" {}
variable "http_auth" {}

terraform {
  required_providers {
    env0 = {
      source = "env0/env0"
    }
    http = {
      source = "hashicorp/http"
      version = "3.4.0"
    }
  }
}

output "api_endpoint" {
  value = local.api_endpoint
}

output "api_key_id" {
  value = var.env0_api_key_id
}

provider "env0" {
  api_key      = var.env0_api_key_id
  api_secret   = var.env0_api_key_value
  api_endpoint = local.api_endpoint
  organization_id = local.kushield_env_org_id
}

data "env0_agent_values" "agent_details" {
  agent_key = local.agent_key
}

output "shag_dev_agent_details" {
  value = data.env0_agent_values.agent_details.values
}

data "http" "agent_details_http" {
  url    = format("https://api-dev.dev.env0.com/agents/%s/values", local.agent_key)
  method = "GET"
  request_headers = {
    accept: "application/json"
    authorization: var.http_auth
    content-type: "application/json"
  }
}

output "http_example_response" {
  value = data.http.agent_details_http.response_body
}

```

It is expected that `output "agent_details"` will print all the agent's details, but it doesn't.
Also, `data "http" "agent_details_http"` which hit the same API EP, does fetch the values, it can be seen by `output "http_example_response"`.

```

Changes to Outputs:
  + api_endpoint           = <...>
  + api_key_id             = <...>
  + http_example_response  = <<-EOT
        "agentKey": "my-agent-key"
        "agentProxy":
          "install": true
        "apiGatewayUrl": <...>
        "awsAccessKeyIdEncoded": <...>
        "awsSecretAccessKeyEncoded": <...>
        "awsSecretsRegion": "us-east-1"
        "cryptoKmsKeyId": <...>
        "deploymentPodWarmPoolSize": 0
        "deploymentResultSqsUrl": <...>
        "deploymentTriggerSqsUrl": <...>
        "env0ApiGwKeyEncoded": <...>
        "eventsTopicArn":<...>
        "httpProxyRequestSqsUrl": <...>
        "httpProxyResponseBucketName": <...>
        "isSelfHosted": "true"
        "limits":
          "cpu": "460m"
          "memory": "1500Mi"
        "outputLogsLogGroupName": <...>
        "prerequisitesStackName": <...>
        "region": "us-east-1"
        "sharedStackPrefix": <...>
        "stage": <...>
        "stateMountPath": <...>
    EOT
  + agent_details = {
      + agent_key = "my-agent-key"
      + id        = "my-agent-key"
      + values    = ""
    }

```

Expected Result / DoD ✅:

"env0_agent_values" works as expected.

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
fixes #690 

### Solution

resty does not play nice with non-struct values.
had to manually handle the string pointer use-case.

### CODE REVIEW

File: client/http/client.go [hunk 1]
```diff
@@ -76,11 +78,22 @@ func (client *HttpClient) Put(path string, request interface{}, response interfa
...
+		responseStrPtr := response.(*string)
+
+		result, err := request.Get(path)
+		if err == nil {
+			*responseStrPtr = string(result.Body())
+		}
+
+		return client.httpResult(result, err)
+	} else {
+		return client.httpResult(request.SetResult(response).Get(path))
```
Comment (TomerHeber) [inline client/http/client.go:95]: normal use-case (regular code) - uses built-in SetResult.

File: client/http/client.go [hunk 2]
```diff
@@ -76,11 +78,22 @@ func (client *HttpClient) Put(path string, request interface{}, response interfa
...
-	result, err := client.request().
-		SetQueryParams(params).
-		SetResult(response).
-		Get(path)
-	return client.httpResult(result, err)
+	request := client.request().SetQueryParams(params)
+
+	responseType := reflect.TypeOf(response)
+
+	if responseType.Kind() == reflect.Ptr && responseType.Elem().Kind() == reflect.String {
```
Comment (TomerHeber) [inline client/http/client.go:85]: if string pointer type (golang reflection).

Comment (amit-alkobi-env0) [reply client/http/client.go:85]: Can you elaborate when each type of response it expected?

Comment (TomerHeber) [reply client/http/client.go:85]: This depends on what the user passes to the function.
If he passes a string pointer - handle special case.
For everything else, use the regular case.

I hope I understood the question. Let me know if it's still unclear.

**Review by amit-alkobi-env0 — APPROVED** [review]

**Pull request merged**